src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -39,9 +39,11 @@
 
 (schema/defn ^:always-validate java-exe-options :- ShellUtils$ExecutionOptions
   [{:keys [env in]} :- ExecutionOptions]
-  (doto (ShellUtils$ExecutionOptions.)
-    (.setEnv (if env (ks/mapkeys name env) {}))
-    (.setStdin in)))
+  (let [exe-options (ShellUtils$ExecutionOptions.)]
+    (.setStdin exe-options in)
+    (when env
+      (.setEnv exe-options (ks/mapkeys name env)))
```

Comment:
Ruby is different because we are running from JRuby and we're intentionally filtering environment variables for the JRubies.  I don't see any reason to mess with the behavior there since we haven't had any issues filed about it.

Yes, the difference in this Clojure code is about passing `nil` vs. passing an empty map.